src/main/java/com/ioet/bpm/people/domain/Person.java

```diff
@@ -26,4 +26,8 @@
     @DynamoDBAttribute
     private String authenticationIdentity;
 
+    @NotBlank
+    @DynamoDBAttribute
+    private String password;
+
```

Comment:
It looks nice, but you also need to modify the save functionality so that to encrypt this value.

Reply:
Do not encrypt passwords and then store the encrypted data. Instead you should do this:

Develop a salt string which is unique to this user. A good place to start would be the timestamp that you are going to be saving with the password + a copy of the user's name string as it existed at the time the password is being saved.

Use sha512 to make a one-way hash of the salt+password.

Save the {salt,hash,timestamp} in a table. Remember in order to enforce a password reuse policy we have to keep the last "n" passwords.

Also some of the "I forgot my password" security/threat models will give a user who has forgotten the current password some "credit" for knowing an older password.

Here is your computer science test:

1. Why do we store a hash instead of encrypting the password?

2. What is the purpose of the salt?

Reply:
👌

Reply:
I added a more secure implementation in how we store passwords, using salts, SHA512 in multiple recursive iterations, with input and design from nspalding.